### ISSUE TITLE
`ty` incorrectly reports `invalid-argument-type` for LangGraph `StateGraph` with `TypedDict` containing `Annotated` fields

### ISSUE THREAD

**Issue by iliadmitriev** [astral-sh/ty#3217]
## Environment
- **ty version**: 0.0.28
- **Python version**: 3.12
- **OS**: macOS (Darwin)
- **LangGraph version**: `>=0.4.0`
- **Command**: `ty check .`

## Description
`ty` reports an `invalid-argument-type` error when passing a standard `typing.TypedDict` to LangGraph's `StateGraph`, claiming it does not satisfy the `TypedDictLikeV1 | DataclassLike` bound. This appears to be a false positive:
1. The class is a standard `typing.TypedDict`
2. LangGraph explicitly supports `TypedDict` with `Annotated` fields for defining state reducers
3. `basedpyright` (v1.33.0) and `pyright` report 0 errors for the exact same code
4. The code executes correctly at runtime

## Reproducible Example

`repro.py`:

```python
from typing import Annotated, TypedDict

from langchain_core.messages import AnyMessage
from langgraph.graph import StateGraph, add_messages

class GraphState(TypedDict):
    messages: Annotated[list[AnyMessage], add_messages]
    iteration_count: Annotated[int, lambda a, b: a + b]

workflow = StateGraph(GraphState)
```

## Steps to Reproduce
1. Create a virtual environment and install dependencies:
   ```bash
   uv venv && source .venv/bin/activate
   uv pip install ty langgraph langchain-core
   ```
2. Save the example above as `repro.py`
3. Run: `ty check repro.py`

## Expected Behavior
No type errors. `TypedDict` should satisfy the `TypedDictLikeV1` bound. Other type checkers (`basedpyright`, `pyright`, `mypy`) accept this pattern without issue, and it's the officially recommended way to define LangGraph state with reducers.

## Actual Behavior
```
error[invalid-argument-type]: Argument to bound method `__init__` is incorrect
   --> repro.py:12:23
    |
 11 | # === Graph Construction ===
 12 | workflow = StateGraph(GraphState)
    |                       ^^^^^^^^^^ Argument type `GraphState` does not satisfy upper bound `TypedDictLikeV1 | DataclassLike` of type variable `StateT`
    |
info: Type variable defined here
  --> langgraph/typing.py:16:1
   |
 16 | StateT = TypeVar("StateT", bound=StateLike)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
info: rule `invalid-argument-type` is enabled by default

Found 1 diagnostic
```

## Additional Context
- This pattern is [officially documented by LangGraph](https://langchain-ai.github.io/langgraph/how-tos/define-state/) for defining graph state with reducers
- The issue likely stems from `ty` not recognizing `TypedDict` subclasses that contain `Annotated` type hints as valid `TypedDictLikeV1` instances
- **Workaround**: Add `# type: ignore[invalid-argument-type]` to the `StateGraph(...)` call
- `basedpyright` with `typeCheckingMode = "recommended"` passes cleanly on the same codebase

### Version

0.0.28

**Comment by carljm** [comment]
Thanks! This looks like #2826
